spec/index.bs

```diff
@@ -1545,6 +1545,14 @@ interface GPUMapMode {
 
   **Returns:** {{ArrayBuffer}}
 
+  1. If |offset| is 0:
+
+    1. Set |offset| to |this|.{{[[mapping_range]]}}[0]
+
+  1. If |size| is 0 and |offset| is greater than or equal to |this|.{{[[mapping_range]]}}[0] and less than |this|.{{[[mapping_range]]}}[1]:
```

Comment:
we'd also need to introduce some "effective mapping offset" thing that's computed off the parent offset and the given nullable offset. For example, see how a similar thing is introduced in another place:
> If size is unspecified, let contentsSize be data.byteLength − dataOffset. Otherwise, let contentsSize be size.

Reply:
I am a bit confused here. Is the offset that user provides w.r.t the `mapping_range` or the buffer?
Validation rules (`offset must be greater than or equal to this.[[mapping_range]][0]`) imply that it's w.r.t buffer, in which case offset should not be calculated from parent offset.

Reply:
it's calculated from the parent offset if it's `null`, which is what my suggestion is here, since `0` is not a good default.
Otherwise, it's from the start of the buffer.